ir/instr.cpp

```diff
@@ -1674,6 +1674,53 @@ unique_ptr<Instr> Malloc::dup(const string &suffix) const {
   return make_unique<Malloc>(getType(), getName() + suffix, *size, isNonNull);
 }
 
+vector<Value*> Calloc::operands() const {
+  return { num, size };
+}
+
+void Calloc::rauw(const Value &what, Value &with) {
+  RAUW(num);
+  RAUW(size);
+}
+
+void Calloc::print(std::ostream &os) const {
+  os << getName() << " = calloc " << *num << " " << *size;
+}
+
+StateValue Calloc::toSMT(State &s) const {
+  auto &[nm, np_num] = s[*num];
+  auto &[sz, np_sz] = s[*size];
+  s.addUB(np_num && np_sz);
+
+  s.addUB(nm.mul_no_uoverflow(sz));
+
+  // TODO: check calloc align.
+  auto p = s.getMemory().alloc(nm * sz, 8, Memory::HEAP);
+
+  s.getMemory().memset(p, { expr::mkUInt(0, 8), true }, nm * sz, 1);
+
+  if (isNonNull)
```

Comment:
calloc is never non-null; there's no C++ throwing equivalent. So this functionality can be removed.